core/services/workflows/engine.go

```diff
@@ -687,6 +692,13 @@ func (e *Engine) finishExecution(ctx context.Context, executionID string, status
 	e.stepUpdatesChMap.remove(executionID)
 	metrics.updateTotalWorkflowsGauge(ctx, e.stepUpdatesChMap.len())
 	metrics.updateWorkflowExecutionLatencyGauge(ctx, executionDuration)
+
+	if executionDuration > fifteenMinutesMs {
```

Comment:
Style nit: you could have just written 15 * time.Minutes :)

Reply:
if i take another pass I will update, thank you!